File: lib/songs/song.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'song.freezed.dart';
part 'song.g.dart';

@freezed
class Song with _$Song {
  factory Song({
    required String id,
    required String title,
    required String artist,
    required String url,
    required int bpm,
    required int unitDuration,
    required List<int> tilesCount,
    required List<int> duration,
    required List<String> tags,
  }) = _Song;

  factory Song.fromJson(Map<String, dynamic> json) => _$SongFromJson(json);
}

File: functions/java/src/main/java/functions/Song.kt
package functions

data class Song(
        val id: String = "",
        val title: String = "",
        val artist: String = "",
        val url: String = "",
        val bpm: Int = 0,
        val unitDuration: Int = 0,
        val tilesCount: List<Int> = listOf(),
        val duration: List<Int> = listOf(),
        val tags: List<String> = listOf()
)
